logrusr/logger.go

```diff
@@ -0,0 +1,143 @@
+package logrusr
+
+import (
+	"fmt"
+	"github.com/activeshadow/logr/util"
+	"github.com/go-logr/logr"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+var (
+	verbosity int
+	loggers   []string
+)
+
+func SetVerbosity(v int) {
+	verbosity = v
+}
+
+func LimitToLoggers(names ...string) {
+	loggers = append(loggers, names...)
+}
+
+type clock struct {
+	mock time.Time // set in tests
+}
+
+func (this *clock) now() time.Time {
+	if this == nil { // normal operations
+		return time.Now()
+	}
+
+	return this.mock // testing only
+}
+
+type LogrusInfoLogr struct {
+	enabled bool
+	name    string
+	kvs     map[string]interface{}
+	logger  logrus.Logger
+
+	// Normally nil, set by test code only.
+	clock *clock
+}
+
+func (this LogrusInfoLogr) Info(msg string, kvs ...interface{}) {
+	if !this.enabled {
+		return
+	}
+
+	logger := this.logger.WithFields(logrus.Fields{
+		"request": &logrus.Fields{
+			"name": this.name,
+			"kvs":  kvs,
+			"this_kvs": this.kvs,
+		},
+	})
+
+	if logger != nil {
+		logger.Info(msg)
+	}
+
+}
+
+func (this LogrusInfoLogr) Enabled() bool {
+	return this.enabled
+}
+
+type LogrusLogr struct {
+	LogrusInfoLogr
+}
+
+func (this LogrusLogr) Error(err error, msg string, kvs ...interface{}) {
+
+	logger := this.logger.WithFields(logrus.Fields{
+		"request": &logrus.Fields{
+			"error": err,
+			"name":  this.name,
+			"kvs":   kvs,
+			"this_kvs": this.kvs,
```

Comment:
Would it make sense to merge key/value pairs? You combine them in the `WithValues` function below, so why not here too?